@xen-orchestra/xva/_writeDisk.mjs

```diff
@@ -13,15 +18,14 @@ export default async function addDisk(pack, vhd, basePath) {
   let written
   let lastBlockWrittenAt = Date.now()
   const MAX_INTERVAL_BETWEEN_BLOCKS = 60 * 1000
-  const chunk_length = 1024 * 1024
-  const empty = Buffer.alloc(chunk_length, 0)
+  const empty = Buffer.alloc(XVA_DISK_CHUNK_LENGTH, 0)
   const stream = await vhd.rawContent()
   let lastBlockLength
   const diskSize = vhd.footer.currentSize
   let remaining = diskSize
   while (remaining > 0) {
-    const data = await readChunkStrict(stream, Math.min(chunk_length, remaining))
-    lastBlockLength = data.length
+    lastBlockLength = Math.min(XVA_DISK_CHUNK_LENGTH, remaining)
+    const data = await readChunkStrict(stream, lastBlockLength)
```

Comment:
What if the stream contains less data than requested? :thinking:

Reply:
readChunkStrict  ensures data.length and lastBlockLength are the same size. Only the last block read may be smaller, and by a known value.

Reply:
My bad, I misread `Math.min` as `Math.max`, it does make sense now :slightly_smiling_face: